compiler/rustc_const_eval/src/interpret/eval_context.rs

```diff
@@ -918,6 +918,7 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         } else {
             self.param_env
         };
+        let param_env = param_env.with_const();
```

Comment:
From past experience with similar regressions I would guess that this causes inlining of `eval_to_allocation` to change, messing up other LLVM optimizations at the caller sites or at other function calls within in this function.

Reply:
Should I add `#[inline]` to these functions or should I remove these calls and assertions altogether?

Reply:
Either profile to see what is now not inlined, or add inline (maybe even always or never) to this function, or any of the functions it calls. I'm not sure what exactly is needed, and only profiling can tell for sure